typescript/src/lib/utils/ledger.ts

```diff
@@ -0,0 +1,177 @@
+import { ethers, Signer } from "ethers"
+import {
+  Account,
+  WalletAPIClient,
+  WindowMessageTransport,
+} from "@ledgerhq/wallet-api-client"
+import BigNumber from "bignumber.js"
+import { AddressZero } from "@ethersproject/constants"
+import { Deferrable } from "@ethersproject/properties"
+import { Hex } from "./hex"
+class AccountNotFoundError extends Error {
+  constructor() {
+    super(
+      "Account not found. Please use `requestAccount` method or set the signer account with `setAccount` method."
+    )
+  }
+}
+
+export class LedgerLiveAppEthereumSigner extends Signer {
+  private _walletApiClient: WalletAPIClient
+  private _windowMessageTransport: WindowMessageTransport
+  private _account: Account | undefined
+
+  constructor(
+    provider: ethers.providers.Provider,
+    windowMessageTransport?: WindowMessageTransport,
+    walletApiClient?: WalletAPIClient
```

Comment:
Actually I don't think it's needed tbh.

In our dApp we also use `@ledgerhq/wallet-api-client-react` package, that also uses `transportMessage` and `walletApiClient` objects and my thinking was that maybe someone would want to use the same objects here and there. I'm using two different ones anyway, but there is one catch to that: in the [wallet api react dosc](https://wallet.api.live.ledger.com/react/configuration) it recommend to just connect the transport and that's it, while in [wallet api core docs](https://wallet.api.live.ledger.com/core/configuration) they recommend to disconnect the transport after the usage. In that case user will have to remember to also disconnect the tranport from react, because if he won't, then the `no ongoingRequest` error appear when the signer will be used for contract communication and the react transport won't be dicsonnected (so two transports will be connected at the same time).

I though that maybe passing those through constructor will make things easier, but I don't think that's the case, because even if user decided to use the same object for both, he will still have to remember to disconnect it after the usage. And using two different intances of it should not be big of a deal.

I've decided to remove both of those arguments in 59e142d20bfa32ff66a4d6779571e5012fc4a58d